std/archive/tar.ts

```diff
@@ -383,7 +387,7 @@ export class Tar {
       fileSize: pad(fileSize, 11),
       mtime: pad(mtime, 11),
       checksum: "        ",
-      type: "0", // just a file
+      type: info?.isDirectory ?? opts.type === "directory" ? "5" : "0",
```

Comment:
Isn’t there an enum that can be used here? The value “5” and “0” are not very descriptive. (I see that it was using “0” before but that seems like bad code. Now would be a good opportunity to clean this up.)

Reply:
I have updated the code, but I am not sure if it is appropriate.

I referenced some websites, and I add "fifo" and "contiguous-file" into Types
> https://github.com/mafintosh/tar-stream
> http://www.gnu.org/software/tar/manual/html_node/Standard.html